domain/schema_validator.go

```diff
@@ -144,28 +122,34 @@ func hasRelatedTableID(relatedTableIDs, updateTableIDs []int64) bool {
 // from usedVer to the latest schema version.
 // NOTE, this function should be called under lock!
 func (s *schemaValidator) isRelatedTablesChanged(currVer int64, tableIDs []int64) bool {
-	if _, ok := s.itemSchemaVers[currVer]; !ok {
+	if len(s.detalItemInfos) == 0 {
+		log.Infof("schema change history is empty, checking %d", currVer)
+		return true
+	}
+	oldestVersion := s.detalItemInfos[0].schemaVersion
+	if versionNewerThan(oldestVersion, currVer) {
```

Comment:
The names are confusing, it doesn't make sense than current version is older than oldestVersion

Reply:
I think we don't need the optimization for a very rare case.